java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -191,6 +192,15 @@ Object applyTransform(Transform transform, Object document ) {
                 .map(JsonPath::compile)
                 .collect(Collectors.toList()));
 
+        if (transform.getApplyOnlyWhen() != null) {
+            Object filterResult = JsonPath.compile(transform.getApplyOnlyWhen()).read(document);
+
+            ArrayList<?> results = (ArrayList<?>) filterResult;
+            if (results != null && results.size() == 0) {
```

Comment:
some form of `CollectionUtils.isEmpty()` - null-safe size check - would make this condition clearer/more readable, if we have such a library method linked here.